### ISSUE TITLE
Drop `unit` and `spectral_axis_unit` keywords to `Spectrum1D` initializer?

### ISSUE THREAD

**Issue by eteq** [astropy/specutils#264]
The title basically says it all.  I can implement this in a PR myself but wanted to first make sure I'm not missing something critical.

To justify it: I'd say the ``unit`` and ``spectral_axis_unit`` keywords are superfluous - if you want a unit on either you need to provide a `Quantity`, which can actually be created in *fewer* characters than ``unit=`` or ``spectral_axis_unit=`` (all that's needed is `` * `` or ``.to``).  Is there some important use case I'm missing by saying that?

(Note also that I think there might be some inconsistencies in what the docstring says and how it actually behaves if you give both a `Quantity` and a ``unit`` keyword.  But I'll only bother to sort that out if the conclusion is that these keywords should *stay* since it's a moot point otherwise.)

cc @nmearl @keflavich @crawfordsm

**Comment by keflavich** [comment]
So you're saying you want:
```
spectral_axis = np.linspace(...)
flux = np.random.randn(...)
Spectrum1D(spectral_axis=spectral_axis, flux=flux, spectral_axis_unit='km/s', unit='Jy')
```
to not be supported, instead requiring
```
spectral_axis = np.linspace(...)*u.km/u.s
flux = np.random.randn(...)*u.Jy
Spectrum1D(spectral_axis=spectral_axis, flux=flux)
```

correct?

I'm mildly opposed to this as I prefer to have more flexibility in the specification.

**Comment by eteq** [comment]
@keflavich - yes that is correct.  My argument is that there is *no* added flexibility in the current situation, but there is  added *complexity*.  That is, compare the following cases (the first is your  first example - current behavior - and the latter is what I'm proposing be standard):
```
Spectrum1D(spectral_axis=spectral_axis, flux=flux, spectral_axis_unit='km/s', unit='Jy')

Spectrum1D(spectral_axis=spectral_axis*u.km/u.s, flux=flux*u.Jy)
Spectrum1D(spectral_axis=u.Quantity(spectral_axis, 'km/s'), flux=u.Quantity(flux, 'Jy'))
```
the latter two convey the same information as the former but with fewer characters *and* without us having to document for the user what to do when the units and various other inputs conflict. As I said before there's already a subtle/non-trivial bug (I think) in that, demonstrating how difficult it might be both to keep it consistent and communicate with users.

I also think this is a better approach *for now*, because it's a lot easier to *add* the unit keywords later if the users demand than take it away if it turns out to confuse them.

**Comment by keflavich** [comment]
I mostly agree, but there is a common use case that is a bit uglier without the keywords

```
Spectrum1D(spectral_axis=u.Quantity(np.linspace(-50,50,51), u.km/u.s),
                    flux=u.Quantity(np.random.randn(51), u.Jy))
```
vs
```
Spectrum1D(spectral_axis=np.linspace(-50,50,51), spectral_axis_unit=u.km/u.s,
                    flux=np.random.randn(51), unit=u.Jy)
```

That said, I now fully agree with you because `spectral_axis_unit` is awkward and `unit` should be `flux_unit` for symmetry reasons.  I hate these keywords.  OK, let's be done with them and require quantities.

**Comment by eteq** [comment]
Cool.  I think this is pretty straightforward so I will try to make a PR to do this in the next week or so, and we can have further discussion there if need be.

**Comment by astrofrog** [comment]
+1 to requiring quantities if units are needed

**Comment by eteq** [comment]
Closed by #277
